Prompt management refactoring and view

sgraband opened the pull request:
#### What it does

- Introduces a new widget to configure AI prompt fragments and system prompts, showing all system prompts, their agents, variants, and customizations.
- Enables viewing and managing active/default variants and customizations with options to revert or delete customizations.
- Displays templates of customizations and unassociated prompt fragments.
- Updates agent handling to use `systemPrompts` with default and active variant logic.
- Refactors prompt fragment management with unified naming:
  - Clear separation between prompt templates (text) and prompt fragments (template + metadata).
  - Support for zero or one built-in versions per fragment and unlimited custom versions.
  - Agents now assign system prompts instead of prompt templates.
  - System prompts consist of variants (1...n), which are prompt fragments grouped under one ID.
  - Adds default and active variant logic for system prompts.
- The prompt service manages all prompt-related requests including customizations and system prompts, with improved state handling and decoupling from the customization service.
- Consumed logic changes in the agent view reflecting default variant handling and system prompt mechanisms.

#### How to test

- Open the prompt fragments configuration widget.
- Verify all system prompts are listed with correct agent associations.
- Check that active and default variants and customizations display correctly.
- Test creating, reverting, and deleting customizations.
- Confirm prompt templates appear correctly for each customization.
- Validate unassociated prompt fragments appear at the end (e.g. create a file in a prompt-templates folder).
- In agent views, verify that system prompts and variant selections behave as expected.
- Check the `#prompt:` autocompletion in the chat.
- Make sure that the agents/ prompt fragments still work as expected.

#### Follow-ups

- Programatically register prompt fragments or variants of system prompts
- Better editing capabilities in view (e.g. create customization in file, copy prompt snippet, ...)
- Render which prompt fragments are used within other prompt fragments
- Annotate the agent with a changed icon in the chat if the non default prompt fragment is used.
- Custom agent prompts should not be registered as built-ins (improve (custom) agent handling)
- Resetting does not always work when there are multiple fragments of the same priority (logic needs to be adapted)

#### Breaking changes

- [x] This PR introduces breaking changes and requires careful review.  
  *Reason:* Changes agent system prompt declaration and prompt service API.

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

sdirix commented on packages/ai-terminal/src/browser/ai-terminal-prompt-template.ts:
```diff
@@ -9,70 +9,72 @@
+    defaultVariant: {
+      id: 'terminal-system-default',
+      template: `{{!-- This prompt is licensed under the MIT License (https://opensource.org/license/mit).
+  Made improvements or adaptations to this prompt template? We’d love for you to share it with the community! Contribute back here:
+  https://github.com/eclipse-theia/theia/discussions/new?category=prompt-template-contribution --}}
+  # Instructions
```
This adds two extra whitespaces at the beginning of every line, right?

sdirix commented on packages/ai-terminal/src/browser/ai-terminal-prompt-template.ts:
```diff
@@ -9,70 +9,72 @@
+    defaultVariant: {
+      id: 'terminal-user-default',
+      template: `{{!-- This prompt is licensed under the MIT License (https://opensource.org/license/mit).
+  Made improvements or adaptations to this prompt template? We’d love for you to share it with the community! Contribute back here:
```
Same whitespace issue here

sdirix commented on packages/ai-ide/src/common/orchestrator-prompt-template.ts:
```diff
@@ -9,42 +9,46 @@
+    defaultVariant: {
+        id: 'orchestrator-system-default',
+        template: `{{!-- This prompt is licensed under the MIT License (https://opensource.org/license/mit).
+    Made improvements or adaptations to this prompt template? We’d love for you to share it with the community! Contribute back here:
```
Same whitespace issue here

sdirix commented on packages/ai-ide/src/common/command-prompt-template.ts:
```diff
@@ -9,213 +9,217 @@
+        id: 'command-system-default',
+        template: `{{!-- This prompt is licensed under the MIT License (https://opensource.org/license/mit).
+    Made improvements or adaptations to this prompt template? We\u2019d love for you to share it with the community! Contribute back here:
+    https://github.com/eclipse-theia/theia/discussions/new?category=prompt-template-contribution --}}
```
These are 4 extra whitespaces at the beginning of every line

sdirix commented on packages/ai-chat/src/common/chat-session-summary-agent-prompt.ts:
```diff
@@ -11,18 +11,21 @@
+    defaultVariant: {
+        id: 'chat-session-summary-prompt',
+        template: `{{!-- !-- This prompt is licensed under the MIT License (https://opensource.org/license/mit).
+    Made improvements or adaptations to this prompt template? We\'d love for you to share it with the community! Contribute back here: ' +
```
This template is broken
![image](https://github.com/user-attachments/assets/99064352-c923-4830-a494-a2a923e88beb)

sdirix reviewed (changes requested):
Generally speaking this is a great change!

My major concern is with the new naming. I provided alternatives for your consideration. Other findings:

## General UI

![image](https://github.com/user-attachments/assets/d33854e9-b26e-4121-bea3-8bb9d04587e1)

## New customizations

New customizations always start empty. This is a regression. Before the refactoring the new customization was autofilled with the previous default.

#### Non-deterministic issues

Changes to the global and local prompt directories, the prompttemplates extensions etc. did not always result in changes in the UI. However I was not able to find a deterministic reproducer. Might be related to file-watcher issues, so not necessarily something to do on your side

sgraband commented:
Thanks for the review @sdirix! I added the proposed renamings, fixed the regression, updated the view a bit and addressed all other feedback comments.

Could you take another look? I also updated the "active variant" to be named selected highlighted it with blue instead of green to distinguish it from the active customization. Let me know what you think.

Imrpovements like better customization creation and changing the active/selected variants will be done in follow ups.

I will update the PR description once approved so that it reflects the renaming changes. Commit message will be updated on squash to reflect the renmaing changes.

sdirix reviewed (approved):
Works for me! Thanks!
